lib/configurator.js

```diff
@@ -27,6 +27,8 @@ function validateOptions (options) {
   )
 }
 
+var defaultCookieDomain = window.location.hostname.split('.').slice(-2).join('.')
```

Comment:
Can’t we use a smarter way using code here? https://github.com/treasure-data/td-js-sdk/blob/master/lib/utils/setCookie.js#L29
-2 doesn’t work well with domains like co.jp and co.uk.

Reply:
@frsyuki Unfortunately, that's used with a feedback cycle to find out if we're setting the cookie on the most permissive domain -- we won't get that sort of feedback loop with a JSONP request.  Or rather, if we tried it, it would be several CDN requests and more costly to do.

Reply:
@MikeMcElroy Don't browsers use the same behavior to check top-level domain for both cookie.domain and Set-Cookie?
I mean code like this:

```javascript
var defaultCookieDomain = (function() {
  var cookie = require('../lib/vendor/js-cookies')
  var domainChunks = document.location.hostname.split('.')
  for (var i = domainChunks.length - 2; i >= 1; i--) {
    var domain = domainChunks.slice(i).join('.')
    var name = "_td_domain_" + domain  // include name to avoid race condition
    cookie.setItem(name, domain, 3600, "/", domain)
    if (cookie.getItem(name) === domain) {
      cookie.removeItem(name, "/", domain)
      return domain
    }
  }
  return document.location.hostname
})()
```

Reply:
i'd be +1 to use something like that

Reply:
@frsyuki I think that would work, but do we need to be worried about user consent since we'll be storing a cookie (even if we don't use it for anything)?  If so, we'd need to calculate this at the time the CDN request goes out, not at startup-time.

Reply:
@dzannotti @MikeMcElroy That’s a good point, I didn’t realize. Yes, it’s better if we evaluate the function (calculate the cookie domain) only when necessary (right before making requests).
I think there're multiple methods to implement it but a way is using null as the default value and fill the value later.